packages/oracle/src/connection-manager.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved
+
+import type { AbstractConnection, ConnectionOptions } from '@sequelize/core';
+import {
+  AbstractConnectionManager,
+  AccessDeniedError,
+  ConnectionError,
+  ConnectionRefusedError,
+  ConnectionTimedOutError,
+  HostNotReachableError,
+  InvalidConnectionError,
+} from '@sequelize/core';
+import { logger } from '@sequelize/core/_non-semver-use-at-your-own-risk_/utils/logger.js';
+import type { Connection as oracledbConnection } from 'oracledb';
+import oracledb from 'oracledb';
+import type { OracleDialect } from './dialect.js';
+
+export type oracledbModule = typeof oracledb;
+
+const debug = logger.debugContext('connection:oracle');
+
+export interface OracleConnection extends oracledbConnection, AbstractConnection {
+  on(event: 'error', listener: (err: any) => void): this;
+}
+
+export interface OracleConnectionOptions extends oracledb.ConnectionAttributes {
+  database?: string;
+
+  host?: string;
+
+  oracleOptions?: object;
+
+  port?: number | string;
+}
+
+export class OracleConnectionManager extends AbstractConnectionManager<
+  OracleDialect,
+  OracleConnection
+> {
+  lib: typeof oracledb;
```

Comment:
should be hard private

```suggestion
  #lib: typeof oracledb;
```

Reply:
Changed to private.